### PR TITLE
Fix GET request on /_synapse/admin/v2/users endpoint

### LINKED ISSUE #6552: /_synapse/admin/v2/users is broken
Running the /_synapse/admin/v2/users as documented produces an internal server error

Postgres logs:
```
STATEMENT:  SELECT name, password_hash, is_guest, admin, user_type, deactivated FROM users WHERE is_guest = false AND deactivated = false ORDER BY name ASC LIMIT 10 OFFSET 0
ERROR:  operator does not exist: smallint = boolean at character 95
```

`\d users` shows that `is_guest` and `deactivated` are type `smallint` not `boolean`.
```

                          Table "public.users"
           Column           |   Type   | Collation | Nullable | Default
----------------------------+----------+-----------+----------+---------
 name                       | text     |           |          |
 password_hash              | text     |           |          |
 creation_ts                | bigint   |           |          |
 admin                      | smallint |           | not null | 0
 upgrade_ts                 | bigint   |           |          |
 is_guest                   | smallint |           | not null | 0
 appservice_id              | text     |           |          |
 consent_version            | text     |           |          |
 consent_server_notice_sent | text     |           |          |
 user_type                  | text     |           |          |
 deactivated                | smallint |           | not null | 0
```

### PR DESCRIPTION
**Pull request by awesome-manuel**
Fixes #6552.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: tests/rest/admin/test_admin.py [hunk 1]
```diff
@@ -341,6 +341,49 @@ def nonce():
...
+class UsersListTestCase(unittest.HomeserverTestCase):
+
+    servlets = [
+        synapse.rest.admin.register_servlets,
+        login.register_servlets,
+    ]
+    url = "/_synapse/admin/v2/users"
+
+    def prepare(self, reactor, clock, hs):
+        self.store = hs.get_datastore()
```
Comment (anoadragon453) [inline tests/rest/admin/test_admin.py]: A store isn't necessary for these tests.

**Comment by wellic** [comment]
@awesome-manuel

I think you forgot to change in code: parse_boolean to parse_integer for guests and deactivated:

guests = parse_integer(request, "guests", default=1)
deactivated = parse_integer(request, "deactivated", default=0)

**Comment by awesome-manuel** [comment]
@wellic you're right, the REST interface should accept both, '0', '1', 'false' and 'true'. Internally we could either use booleans or ints. Database uses ints already but I'd prefer booleans for internal interfaces.

**Comment by richvdh** [comment]
> @richvdh Your suggestion was to use '0' and '1' on the REST interface, right?

uhhh I don't think so... `true` and `false` seem better for boolean values, even if they are represented as ints in the database.

**Comment by awesome-manuel** [comment]
> > @richvdh Your suggestion was to use '0' and '1' on the REST interface, right?
> 
> uhhh I don't think so... `true` and `false` seem better for boolean values, even if they are represented as ints in the database.

@wellic @anoadragon453 then I think the code is already correct.
The API still returns '0' and '1' in the JSON, but this would be another change.

**Pull request merged**